### DISCUSSION TITLE
Screen Resolution & Responsive (mobile) design

### DISCUSSION THREAD

**Discussion by alexanderson1993** [Thorium-Sim/thorium-nova#44] [Ideas]
Just throwing this out there before I forget - what screen resolution will we develop towards?

Thorium Classic was designed for 1600x900 screen resolution, but most cards had the ability to scale down to 1024x768 and up to 1080p. Thorium Classic didn't support mobile devices at all. Instead, there was a separate React Native app that connected to Thorium Classic with a few pared-down screens.

My preference for Thorium Nova is to only design for 1080p screen, but I'd love to get everyone else's takes.

Also, I have absolutely no idea how to approach mobile screens. There's just a lot of stuff that goes into these controls, so making every card mobile responsive seems a monumental design task. Suggestions for this point are welcome.

**Comment by alexanderson1993** [comment]
One other thing that needs to be considered: Touchscreen and multi-touch capability. More and more, I'm seeing people purchase touchscreens for their bridges. We really need to make sure the UIs work well with touchscreens and multitouch. This includes designing the screens so multitouch works well, and not putting two things you would likely control at the same time right on top of each other.

**Reply by astrolamb-gaming** [reply]
I'd like to see mobile screens be able to be used, but maybe we should decide on which consoles/cards _should_ work on mobile and only bother with making others work later on, if at all.
I don't have a particular preferrence regarding resolution.

**Comment by Xansta** [comment]
I suggest we create cards (the building blocks for screens or consoles) that can easily scale to a range of resolutions. The specialty screens like the viewscreen or others that might have a specific role in a physical bridge, we might constrain to a particular resolution.

One of the things that drew me to Artemis in the first place was that it was available on iOS. We played many games using version 2.4.0 for that reason (later versions were not backward compatible). I like Empty Epsilon's offering on Android devices (no iOS yet) for similar reasons.

Most of my SBS interaction occurs over the Internet. I think the clients need to be as resolution flexible as possible. If we need to place constraints on resolution, we should set a range. Something else to consider: portrait vs. landscape - monitors are typically landscape, many tablets and phones lean towards portrait.

EE was designed around touchscreens. Artemis can work with touch screens with some manipulation. If Nova supports touchscreen from the ground up, it's easier to support mobile devices.

**Reply by Froginator45** [reply]
In the HMI industry, I have found that anything less than 1080p is becoming less and less common. 1080p tends to be the standard with a few clients even having higher resolutions (especially for larger view screens (up to 4k)). If everything is vector graphics then the 1080p scaling up should generally be fine for the larger screens. We have found that designing a screen for both mobile and desktops can be very difficult in most cases. Generally you have one of two situations: overly cluttered (and unusable) interface on mobile or overly simplified and frustrating interface on desktop. We tend to have to design mobile screens separate from desktop screens. With the mobile versions being much more specific to the situation. Some cards might also only make sense on mobile (i.e. handheld scanner). Others on desktop (i.e. exterior view screens).

As far as touch screens go, I can see them being very useful for lots of the cards... (navigation, medical, scanners, engineering, etc.)

**Reply by Xansta** [reply]
One of the things that draws me to the SBS is the relatively low hardware requirements. I am able to use laptops or desktops that would otherwise be considered obsolete for consoles in a starship bridge simulator. Maybe I am the exception rather than the rule nowadays - most folks may have much nicer hardware than I do. Many of the screens on these laptops are not 1080P. Many have low end video requiring 1024x768 for effective use. I know several users of Empty Epsilon that run Raspberry Pis as consoles.

I'm not saying we should dumb everything down to the lowest hardware spec, but there should be some acknowledgement of users with lower hardware being included as part of a crew. We should provide guidance as to performance expectations of each card/screen/console.

**Reply by alexanderson1993** [reply]
My approach to lower-powered devices is to have "Lite" cards that have lower requirements. For example, good luck running WebGL on a Raspberry Pi, but a lot of the screens in Thorium Nova will have that requirement, since space is 3D and all.

So the idea is that there is a set of screens that is able to run really well on those lower-powered devices, and maybe they are also designed to work great on 4:3 aspect ratio screens.
